### PR TITLE
- empty values now are parsed as unset in range

### PR DESCRIPTION
**Pull request by mark-tate**
- date format now can be passed falsy values as well as strings
- updates tests
- updates stories
- updates docs

### CODE REVIEW

File: packages/lab/src/__tests__/__e2e__/date-input/DateInputRange.cy.tsx [hunk 1]
```diff
@@ -58,70 +67,54 @@ function assertDateChange(
...
-    expect(date.startDate).to.be.undefined;
-  } else if (expectedValidStartDate) {
-    // assert valid start date matches expected date
+  if (expectedValidStartDate) {
+    // assert valid start date
     expect(date.startDate).not.to.be.undefined;
     expect(adapter.format(date.startDate, "DD MMM YYYY")).to.equal(
       adapter.format(expectedDate.startDate, "DD MMM YYYY"),
     );
+  } else if (expectedDate?.startDate === undefined) {
```
Comment (origami-z) [inline packages/lab/src/__tests__/__e2e__/date-input/DateInputRange.cy.tsx:76]: Feels range's `undefined` has slightly different meaning than single date's `Date | null | undefined`...

**Review by origami-z — APPROVED** [review]
We can release this in labs and iterate further

**Comment by origami-z** [comment]
Changed target to `main` branch, so we can get a preview

File: packages/lab/src/date-input/DateInputSingle.tsx [hunk 2]
```diff
@@ -160,17 +161,17 @@ export const DateInputSingle = forwardRef<
...
       onClick,
       emptyReadOnlyMarker = "—",
       endAdornment,
-      format = "DD MMM YYYY",
       inputProps = {},
       inputRef: inputRefProp = null,
-      locale,
       parse: parseProp,
-      placeholder = "dd mmm yyyy",
+      placeholder = format.toLowerCase(),
```
Comment (origami-z) [inline packages/lab/src/date-input/DateInputSingle.tsx:174]: lower case and upper case in date formatter are different. A few suggestions
- Remove, leave empty (no other Salt component sets this)
- Set as generic text, e.g. "Set Date", "Start Date", etc.
- Or leave as `format`

Comment (mark-tate) [reply packages/lab/src/date-input/DateInputSingle.tsx:174]: @ivan-calderon / @jake-costa / @honey-chang  one for you to opine on..
Feels like this falls into the formatting pattern, an overall guide on recommended formats?
Simple change to make once we agree on what it should be (compare with UITK?)

**Review by origami-z — APPROVED** [review]

**Pull request merged**